core/src/main/scala/org/apache/spark/sql/TiExtensions.scala

```diff
@@ -34,14 +38,20 @@ class TiExtensions extends (SparkSessionExtensions => Unit) {
 
   // call from pyspark only
   def getOrCreateTiContext(sparkSession: SparkSession): TiContext =
-    synchronized {
-      tiContextMap.get(sparkSession) match {
-        case Some(tiContext) => tiContext
-        case None =>
-          // TODO: make Meta and RegionManager independent to sparkSession
-          val tiContext = new TiContext(sparkSession)
-          tiContextMap.put(sparkSession, tiContext)
-          tiContext
+    // check catalog plugin mode
+    if (!TiExtensions.validateCatalog(sparkSession)) {
+      logger.error("TiSpark is not work in catalog plugin mode!")
```

Comment:
The description will cause misunderstanding, be more clear.
Don't use is, use model verbs such as "should, must".
And also add some description on how to resolve.